apps/web/src/components/providers/DocsProvider.tsx

```diff
@@ -0,0 +1,192 @@
+  const [path, setPath] = useState<string>('');
+  const [open, setOpen] = useState(false);
+  const [voted, setVoted] = useState('');
+  const { pathname } = useLocation();
+  const segment = useSegment();
+  const { colorScheme } = useMantineColorScheme();
+  const isDark = colorScheme === 'dark';
+
+  const onClose = () => {
+    toggle();
+  };
+
+  useEffect(() => {
+    return () => {
+      setPath('');
+      setVoted('');
+    };
+  }, [pathname]);
+
+  const toggle = () => {
+    setOpen(!open);
+  };
+
+  const onVoteClick = (vote: string) => () => {
+    if (voted.length > 0) {
+      return;
+    }
+    segment.track('Inline docs voting used', {
+      documentationPage: path,
+      pageURL: window.location.href,
+      vote,
+    });
+    setVoted(vote);
+  };
+
+  return (
+    <DocsContext.Provider
+      value={{
+        setPath,
+        toggle,
+        enabled: path.length > 0,
+        path,
+      }}
+    >
+      <Modal
+        opened={open}
+        onClose={onClose}
+        styles={{
+          root: {
+            zIndex: 10003,
+          },
+          inner: {
+            padding: '24px',
+          },
+          body: {
+            height: '100%',
+            maxHeight: '100%',
+          },
+          modal: {
+            width: 800,
+            padding: '24px !important',
+            borderRadius: 12,
+            position: 'relative',
+            height: '100%',
+          },
+        }}
+        overflow="inside"
+        withCloseButton={false}
+        overlayColor={isDark ? colors.BGDark : colors.BGLight}
+      >
+        <Docs path={path}>
+          <Flex
+            className={css({
+              position: 'fixed',
+              top: '150',
+              right: '150',
+              background: isDark ? 'legacy.B15' : 'white',
+              zIndex: 1,
+              padding: '25',
+              borderBottomLeftRadius: '50',
+            })}
+            gap="125"
+          >
+            <Tooltip label="Open docs website">
+              <ActionIcon
+                style={{
+                  width: '20px  !important',
+                  minWidth: '20px !important',
+                  border: 'none',
+                }}
+                variant="transparent"
+                onClick={() => {
+                  window.open(`https://docs.novu.co/${path}`);
+                }}
+              >
+                <IconOpenInNew />
+              </ActionIcon>
+            </Tooltip>
+            <ActionIcon
+              variant="transparent"
+              onClick={onClose}
+              style={{
+                width: '20px  !important',
+                minWidth: '20px !important',
+                border: 'none',
+              }}
+            >
+              <IconOutlineClose />
+            </ActionIcon>
+          </Flex>
+        </Docs>
+        <Flex
+          className={css({
+            marginTop: '250',
+          })}
+          align="center"
+          gap="125"
+        >
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Please hide this if the documentation fails to load -- otherwise, we'll collect some false signals‏